forte/data/base_pack.py

```diff
@@ -515,44 +516,47 @@ def entry_getter(cls: Entry, attr_name: str):
             if attr_type[0] in ENTRY_TYPE_DATA_STRUCTURES:
                 # Generate FList/FDict object on the fly
                 return attr_type[0](parent_entry=cls, data=attr_val)
-            try:
-                # Check dataclass attribute value type
-                # If the attribute was an Entry object, only its tid
-                # is stored in the DataStore and hence its needs to be converted.
-
-                # Entry objects are stored in data stores by their tid (which is
-                # of type int). Thus, if we enounter an int value, we check the
-                # type information which is stored as a tuple. if any entry in this
-                # tuple is a subclass of Entry or is a ForwardRef to another entry,
-                # we can infer that this int value represents the tid of an Entry
-                # object and thus must be converted to an object using get_entry
-                # before returning.
-                if (
-                    isinstance(attr_val, int)
-                    and attr_type[1]
-                    and any(
-                        issubclass(entry, Entry)
-                        if isclass(entry)
-                        else is_forward_ref(entry)
-                        for entry in list(attr_type[1])
-                    )
-                ):
-                    return cls.pack.get_entry(tid=attr_val)
-
-                # The condition below is to check whether the attribute's value
-                # is a pair of integers - `(pack_id, tid)`. If so we may have
-                # encountered a `tid` that can only be resolved by
-                # `MultiPack.get_subentry`.
-                elif (
-                    isinstance(attr_val, tuple)
-                    and len(attr_val) == 2
-                    and all(isinstance(element, int) for element in attr_val)
-                    and hasattr(cls.pack, "get_subentry")
-                ):
-                    # Multi pack entry
-                    return cls.pack.get_subentry(*attr_val)
-            except KeyError:
-                pass
+            elif attr_type[1] == (FNdArray,):
+                # Generate FNdArray object on the fly
+                return FNdArray(data_ref=attr_val)
```

Comment:
Maybe this is the reason that the interface of `FNdArray` is a bit weird? Can we use kwargs to support this?